phoenix_liveview/phoenix_liveview.exs

```diff
@@ -0,0 +1,134 @@
+Application.put_env(:phoenix, :json_library, Jason)
+Application.put_env(:esbuild, :version, "0.12.18")
+
+Mix.install(
+  deps = [
+    {:phoenix, "~> 1.6.0-rc.0", override: true},
+    :jason,
+    :plug_cowboy,
+    {:phoenix_live_view, "~> 0.16.0"},
+    {:esbuild, "~> 0.2"}
+  ]
+)
+
+Code.require_file("mix_install_deps_backport.exs")
+
+Application.put_env(:esbuild, :default,
+  args: ~w(app.js --bundle --target=es2016 --outdir=../priv/static/assets),
+  cd: Path.expand("assets", __DIR__),
+  env: %{"NODE_PATH" => InstallFolderTemporaryBackport.determine_build_folder(deps) <> "/deps"}
+)
+
+Application.put_env(:live_clock, LiveClock.Endpoint,
+  server: true,
+  http: [ip: {127, 0, 0, 1}, port: 8081],
+  secret_key_base: "vuLgz/lXn+03HJIPTHbTMeZGd16UzvFxLgThphnLdafNmlZqCSGEZJe3Hp9cRhVs",
+  live_view: [signing_salt: "xF2dLhep"],
+  debug_errors: true,
+  watchers: [
+    esbuild: {Esbuild, :install_and_run, [:default, ~w(--sourcemap=inline --watch)]}
+  ]
+)
+
+defmodule LiveClock.ClockLive do
+  use Phoenix.LiveView
+  use Phoenix.HTML
+  import Phoenix.LiveView.Helpers
+
+  def render(assigns) do
+    ~L"""
+    <div id="svg-container">
+      <svg id="clock" width="600" height="650">
+        <g transform="scale(3)">
+          <circle style="fill: #9ab" cx="102" cy="102" r="83"/>
+          <circle style="fill: #666" cx="100" cy="100" r="83"/>
+          <circle style="fill: #FFF" cx="100" cy="100" r="77"/>
+          <%= for h <- (0..59) do %>
+          <line style="stroke: #eee; stroke-width: 1px" x1="100" y1="28" x2="100" y2="34" transform="rotate(<%= 360 * h / 60.0 %> 100 100)"/>
+          <% end %>
+          <%= for h <- (0..11) do %>
+          <line style="stroke: #aaa; stroke-width: 1px" x1="100" y1="28" x2="100" y2="35" transform="rotate(<%= 360 * h / 12.0 %> 100 100)"/>
+          <% end %>
+          <line style="stroke: #888; stroke-width: 3px" x1="100" y1="100" x2="100" y2="50"
+            transform="rotate(<%= 360 * (@date.hour + @date.minute / 60.0) / 12.0 %> 100 100)"/>
+          <line style="stroke: #888; stroke-width: 3px" x1="100" y1="100" x2="100" y2="27"
+            transform="rotate(<%= 360 * (@date.minute + @date.second / 60.0) / 60.0 %> 100 100)"/>
+          <line style="stroke: #E88; stroke-width: 1px" x1="100" y1="100" x2="100" y2="27"
+            transform="rotate(<%= 360 * @date.second / 60.0 %> 100 100)"/>
+        </g>
+        <text x="130" y="585" font-family="cursive" font-size="17" style="fill: #789">
+           The "almost-one-file" Phoenix LiveView clock app
+        </text>
+      </svg>
+    </div>
+    """
+  end
+
+  def mount(_params, _session, socket) do
+    if connected?(socket), do: :timer.send_interval(100, self(), :tick)
+
+    {:ok, put_date(socket)}
+  end
+
+  def handle_info(:tick, socket) do
+    {:noreply, put_date(socket)}
+  end
+
+  defp put_date(socket) do
+    assign(socket, date: NaiveDateTime.local_now())
```

Comment:
`NaiveDateTime.local_now()` does not currently support fractional seconds, otherwise I would have used that to make a smoother (sub-second) needle move.